src/mito2/src/worker/handle_manifest.rs

```diff
@@ -74,7 +75,7 @@ impl RegionEditQueue {
     }
 }
 
-impl<S> RegionWorkerLoop<S> {
+impl<S: LogStore> RegionWorkerLoop<S> {
```

Comment:
Maybe we can only add the trait bound to `handle_manifest_region_change_result()`